azurerm/resource_arm_log_profile.go

```diff
@@ -0,0 +1,259 @@
+		Read:   resourceArmLogProfileRead,
+		Update: resourceArmLogProfileCreateOrUpdate,
+		Delete: resourceArmLogProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"storage_account_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: azure.ValidateResourceIDOrEmpty,
+			},
+			"service_bus_rule_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: azure.ValidateResourceIDOrEmpty,
+			},
+			"locations": {
+				Type:     schema.TypeList,
+				MinItems: 1,
+				Required: true,
+				Elem: &schema.Schema{
+					Type:             schema.TypeString,
+					StateFunc:        azureRMNormalizeLocation,
+					DiffSuppressFunc: azureRMSuppressLocationDiff,
+				},
+			},
+			"categories": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type:             schema.TypeString,
+					DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				},
+			},
+			"retention_policy": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"days": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceArmLogProfileCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).logProfilesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	categories := expandLogProfileCategories(d)
+	locations := expandLogProfileLocations(d)
+	retentionPolicy, err := expandAzureRmLogProfileRetentionPolicy(d)
+	if err != nil {
+		return err
+	}
+
+	logProfileProperties := insights.LogProfileProperties{
+		Categories:      &categories,
+		Locations:       &locations,
+		RetentionPolicy: retentionPolicy,
+	}
+
+	storageAccountID := d.Get("storage_account_id").(string)
+	if storageAccountID != "" {
+		logProfileProperties.StorageAccountID = utils.String(storageAccountID)
+	}
+
+	serviceBusRuleID := d.Get("service_bus_rule_id").(string)
+	if serviceBusRuleID != "" {
+		logProfileProperties.ServiceBusRuleID = utils.String(serviceBusRuleID)
+	}
+
+	name := d.Get("name").(string)
+	parameters := insights.LogProfileResource{
+		Name:                 utils.String(name),
+		LogProfileProperties: &logProfileProperties,
+	}
+
+	_, createErr := client.CreateOrUpdate(ctx, name, parameters)
+	if createErr != nil {
+		return fmt.Errorf("Error Creating/Updating Log Profile %q: %+v", name, createErr)
+	}
+
+	// Wait for Log Profile to become available
+	err = resource.Retry(300*time.Second, retryLogProfilesClientGet(name, meta))
```

Comment:
*question* Is the `Retry` here required for this resource?

Reply:
Yup, the client returns a few 404s before eventually returning a 200.

Reply:
Hmm, that might be some API issue. Do we have bugs created in their [repository](https://github.com/Azure/azure-rest-api-specs)?

---
In reply to: [212493661](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1792#discussion_r212493661) [](ancestors = 212493661)